### PR TITLE
Ft set roles permission

### PR DESCRIPTION
**Pull request by UmuhireJessie**
### What does this PR do?
- CRUD operation on role, permissions
- Assign permissions to a role
- Assign a user a particular role in the Database
### Description of Task to be completed?
- Documenting API
- Testing API
### How should this be manually tested?
`npm run dev` to start the server manually
use postman to test different endpoints present in [`index.js`](https://github.com/atlp-rwanda/Dominators-Phantom-BE/blob/develop/src/routes/index.js)
### Any background context you want to provide?
Use of permissions that are in the database while checking if a logged-in user is permitted to do an operation. Permissions used are in an object for now.
### Questions:

### CODE REVIEW

**Review by furebo — COMMENTED** [review]
Please rebase to resolve the conflicts

**Review by furebo — COMMENTED** [review]
Create also seeders for roles

**Review by fab-ryan — APPROVED** [review]

**Comment by Anguandia** [comment]
Good work @UmuhireJessie, please do the following:
- respond to review comments
- give more context in describing how to test this feature
- since we are not using pivotal tracker, you may just include a link to the Trello card for this feature in the section _relevant pivotal tracker stories_
- since you have no screenshots included, you may as well exclude that heading

File: src/routes/index.js [hunk 1]
```diff
@@ -2,10 +2,16 @@ import express from 'express';
...
+import roleRoutes from './api/roleRoutes';
+import rolePermissionRoutes from './api/rolepermissionRoutes';
 
 const routes = express.Router();
 routes.use('/users', userRoutes);
 routes.use('/buses', crudBusRoutes);
 routes.use('/routes', busRoutes);
+routes.use('/permissions', permissionRoutes)
+routes.use('/roles', roleRoutes)
+routes.use('/roles', rolePermissionRoutes)
```
Comment (tjeaneric) [inline src/routes/index.js:20]: /roles in two different middlewares

**Review by tjeaneric — CHANGES_REQUESTED** [review]
Rebase and Solve Conflicts

**Comment by tjeaneric** [comment]
Squash commits

**Review by fab-ryan — COMMENTED** [review]
Resolve the conflicts

**Review by GilbertUp — APPROVED** [review]

**Comment by UmuhireJessie** [comment]
> Create also seeders for roles

Done

**Comment by UmuhireJessie** [comment]
Made a new branch for easing the work of squashing. Its name is `ft-new-set-roles-permission`. Please review it and make a suggestions if any.

**Pull request closed without merging**